examples/chatbot_app.py

```diff
@@ -0,0 +1,120 @@
+import streamlit as st
+from chatbot import chatbot, extract_instructions
+import os
+import json
+import time
+import logging
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+email = os.getenv('EMAIL')
+
+st.header("Chatbot")
+
+st.markdown("""
+    <style>
+    .message-box {
+        max-width: 40%;
+        padding: 10px;
+        border: 1px solid #ccc;
+        border-radius: 5px;
+        margin: 10px 0;
+    }
+    .user-message {
+        background-color: #D3D3D3;
+        text-align: left;
+    }
+    .bot-message {
+        background-color: #D3D3D3;
+        text-align: right;
+        margin-left: auto;
+    }
+    </style>
+    """, unsafe_allow_html=True)
+
+openai_api_key = st.text_input("Enter OpenAI API Key:", type="password")
+api_key = st.text_input("Enter API Key:", type="password")
+
+if 'conversation' not in st.session_state:
+    st.session_state.conversation = []
+
+for message in st.session_state.conversation:
+    if message["role"] == "user":
+        st.markdown(f"<div class='message-box user-message'><strong>User:</strong> {message['content']}</div>", unsafe_allow_html=True)
+    else:
+        st.markdown(f"<div class='message-box bot-message'><strong>Chatbot:</strong> {message['content']}</div>", unsafe_allow_html=True)
+
+user_query = st.text_area("User Query", height=100)
+system_prompt = st.text_area("System Prompt", height=100)
+
+if st.button("Chat with Chatbot"):
+    if not user_query.strip():
+        st.write("Please enter a user query.")
+    elif not openai_api_key.strip() or not api_key.strip():
+        st.write("Please enter both API keys.")
+    elif not system_prompt.strip():
+        st.write("Please enter a system prompt.")
+    else:
+        start_time = time.time() 
+
+        user_message = {"role": "user", "content": user_query}
+        st.session_state.conversation.append(user_message)
+
+        instructions = extract_instructions(system_prompt)
+        st.session_state["system_prompt"] = system_prompt
+
+        try:
+            (
+                chatbot_response, hallucination_score, toxicity, conciseness, 
+                adherence_details, completeness, detection_response
+            ) = chatbot(user_query, instructions, openai_api_key, api_key)
+        except Exception as e:
+            st.write(f"Error occurred: {e}")
+            chatbot_response = "Error occurred while processing your request."
+
+        end_time = time.time()
+        elapsed_time = end_time - start_time
+        logging.info(f"Time taken for response: {elapsed_time:.2f} seconds")    
+
+        if chatbot_response == "Error occurred while processing your request.":
+            st.write(chatbot_response)
+        else:
+            st.session_state['hallucination_score'] = hallucination_score
```

Comment:
where are these objects (hallucination_score, toxicity etc.) that are stored within `st.session_state` used?

Reply:
these objects aren't used anymore